### PR TITLE
zsh: Create package zsh-powerlevel9k

### PR DESCRIPTION
**Pull request by pierrechevalier83**
To use, add this option to your configuration.nix:
`programs.zsh.promptInit = "source ${pkgs.zsh-powerlevel9k}/share/zsh-powerlevel9k/powerlevel9k.zsh-theme";`

###### Motivation for this change
I want to be able to use the powerlevel9k prompt without all the bloat of oh-my-zsh or other

###### Things done
Created package zsh-powerlevel9k that can be enabled by overriding the zsh.promptInit variable

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/programs/zsh/zsh.nix [hunk 1]
```diff
@@ -89,8 +89,8 @@ in
         description = ''
           Enable zsh-autosuggestions
         '';
+        type = types.bool;
```
Comment (Mic92) [inline nixos/modules/programs/zsh/zsh.nix]: please make this a separate commit.

File: pkgs/shells/zsh-powerlevel9k/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, zsh }:
+
+# To make use of this derivation, use
+# `programs.zsh.promptInit = "source ${pkgs.zsh-powerlevel9k}/share/zsh-powerlevel9k/powerlevel9k.zsh-theme";`
+
+stdenv.mkDerivation rec {
+  name = "powerlevel9k-${version}";
+  version = "master";
```
Comment (Mic92) [inline pkgs/shells/zsh-powerlevel9k/default.nix]: We use the following scheme for projects without releases:

```diff
- version = "master";
+ version = "2017-11-10";
```

File: nixos/modules/programs/zsh/zsh.nix (hunk 1 above)
Comment (pierrechevalier83) [reply nixos/modules/programs/zsh/zsh.nix]: Thanks for the review, Mic92. I've just done that.

**Comment by Mic92** [comment]
I merged both with commit messages adapted to our naming conventions.
